include/inviwo/core/properties/buttongroupproperty.h

```diff
@@ -0,0 +1,105 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <inviwo/core/common/inviwocoredefine.h>
+#include <inviwo/core/common/inviwo.h>
+
+#include <inviwo/core/properties/property.h>
+#include <functional>
+#include <optional>
+
+namespace inviwo {
+
+/**
+ * \ingroup properties
+ * \brief The Button Group property class provides a group of button that you can bind functions to.
+ * The button group property has a widget witch creates a row of buttons and register functions to
+ * them.
+ * @see ButtonGroupPropertyWidgetQt
+ */
+class IVW_CORE_API ButtonGroupProperty : public Property {
```

Comment:
Well it is not a property owner, just a set of buttons.

Reply:
I see several potential problems with using a special property for multiple buttons instead of basing it on a CompositeProperty and using the existing ButtonProperty:
- Duplication of behaviour (the same functionality exist in ButtonProperty)
- Special behaviour required for linking (converters)/python/javascript

I therefore think that incorporating layout capabilities would be a better approach as it would be consistent with other code.

Reply:
I don't think that is a good idea since it mixes UI layouting stuff with  functionality.

Reply:
@martinfalk Is that not exactly what ButtonGroup does though? The ButtonGroup is just for layouting?

Reply:
If someone wants to have a unique layout for one of their properties they should make their own widget for it. For example, if someone wants to align buttons on a row, they should make their own widget for that. (As Peter has done here)

Reply:
This is what we have the property semantic for, i.e., different views of the same property. So we could create a horizontal layout semantic for the CompositeProperty. No new ButtonGroup property needed. Just as you say a widget to display buttons in a row

Reply:
But how would we layout Properties that are not ButtonProperties in a horizontal layout? IMO it only makes sense for button properties, hence it should not be a semantic for CompositeProperty

Reply:
Our FileProperty could have used such a feature for example, e.g., input field and a button. 

I don’t see a major reason to re-implement button behaviour because there are multiple ones. It disrupts other existing functionality such as linking.

Reply:
I guess ButtonGroupProperty could derive from CompositeProperty instead of property and adds it buttons and button properties. But instead of using CompositeProperty's widget it still uses its own widget.

Reply:
Agree

Reply:
I will never implement that. I you want to have a try at layouting our properties. which btw are all designed to all be layouted vertically, in a horiontal layout go ahead. But that is going to be a monumental effort and one that will only very special cases. And which probably will not look half as good as what we have now. And for what use? To redo the fileproperty? I have better things to do.

Reply:
It should be quite a small effort to change such that ButtonGroupProperty inherit from CompositeProperty and use existing button properties instead of new custom ones? The widget would remain unchanged...

Reply:
no.

Reply:
Thought about it some more. Making ButtonGroupProperty derive from CompositeProperty would make it possible for anyone to add additional properties to it which, IMO makes it a no go. 
Maybe possibly it could work with private inheritance of either CompositeProperty or PropertyOwner and only make public what is needed for the bindings. But I dont have a good feeling about such a solution, it does not feel stable and future proof.

Reply:
I would say all of this is beyond this PR. I opt for leaving the buttons as they are.

Reply:
:+1: